File: solutions/python/flower-field/1/flower_field.py
def annotate(garden):
    if not garden:
        return garden

    row_length = len(garden[0])

    for row in garden:
        # validar longitud
        if len(row) != row_length:
            raise ValueError("The board is invalid with current input.")

        # validar caracteres
        for char in row:
            if char not in (" ", "*"):
                raise ValueError("The board is invalid with current input.")
    result = []
    for i in range(len(garden)):
        row = garden[i]
        temp = ""
        for j in range(len(row)):
            if row[j] == " ":
                mines = _count_flowers(i,j,garden)
                temp += str(mines) if mines > 0 else " "
            else:
                temp += row[j]
        result.append(temp)
    return result


def _count_flowers(row, col, garden):
    mines = 0
    rows = len(garden)
    directions = [[-1, 0], [1, 0], [0, -1], [0, 1], [-1, -1], [-1, 1], [1, -1], [1, 1]]
    for direction in directions:
        new_row = row + direction[0]
        new_col = col + direction[1]
        if 0 <= new_row < rows:
            near_row = garden[new_row]
            if 0 <= new_col < len(near_row) and near_row[new_col] == "*":
                mines += 1
    return mines

File: solutions/cpp/flower-field/1/flower_field.cpp
#include "flower_field.h"

namespace flower_field {
    static const int directions[8][2] = {
        {-1, 0}, {1, 0},
        {0, -1}, {0, 1},
        {-1, -1}, {-1, 1},
        {1, -1}, {1, 1}
    };

    static int count_mines(const size_t row, const size_t column, const std::vector<std::string>& garden) {
        int mines = 0;
        for (const auto direction : directions) {
            const size_t new_row = row + direction[0];
            const size_t new_column = column + direction[1];
            if (new_row < garden.size()) {
                const std::string& new_garden = garden[new_row];
                if (new_column < new_garden.size() && new_garden[new_column] == '*') {
                    mines++;
                }
            }
        }
        return mines;
    }

    std::vector<std::string> annotate(const std::vector<std::string> &flower_field) {
        std::vector<std::string> garden = flower_field;
        for (size_t i = 0; i < flower_field.size(); i++) {
            for (size_t j = 0; j < flower_field[0].size(); j++) {
                if (garden[i][j] == ' ') {
                    const int mines = count_mines(i,j,garden);
                    garden[i][j] = mines > 0 ? static_cast<char>('0' + mines): ' ';
                }
            }
        }

        return garden;
    }
} // namespace flower_field
